### PR TITLE
gs: Include antenna location in forwarded message RxMetadata

### PR DESCRIPTION
**Pull request by neoaggelos**
#### Summary

Required for #1815 

#### Changes

- Gateway Server properly adds antenna location on the `RxMetadata` field of uplink messages forwarded to the Network Server

#### Notes for Reviewers

- Implementing this as a separate small fix.

#### Checklist

- [x] Scope: The referenced issue is addressed, there are no unrelated changes.
- [X] Compatibility: The changes are backwards compatible with existing API, database and configuration, according to the stability commitments in `README.md`.
- [x] Testing: The changes are covered with unit tests. The changes are tested manually as well.
- [x] Documentation: Relevant documentation is added or updated.
- [ ] Changelog: Significant features, behavior changes, deprecations and fixes are added to `CHANGELOG.md`.
- [X] Commits: Commit messages follow guidelines in `CONTRIBUTING.md`, there are no fixup commits left.

### CODE REVIEW

File: pkg/gatewayserver/io/io.go [hunk 1]
```diff
@@ -159,6 +159,11 @@ func (c *Connection) HandleUp(up *ttnpb.UplinkMessage) error {
 		}
 		md.UplinkToken = buf
 		md.DownlinkPathConstraint = c.gateway.DownlinkPathConstraint
+
+		if len(c.gateway.Antennas) > int(md.AntennaIndex) {
+			location := c.gateway.Antennas[md.AntennaIndex].Location
+			md.Location = &location
```
Comment (johanstokking) [inline pkg/gatewayserver/io/io.go]: Ineffective variable

File: pkg/gatewayserver/io/io.go [hunk 2]
```diff
@@ -159,6 +159,11 @@ func (c *Connection) HandleUp(up *ttnpb.UplinkMessage) error {
 		}
 		md.UplinkToken = buf
 		md.DownlinkPathConstraint = c.gateway.DownlinkPathConstraint
+
+		if len(c.gateway.Antennas) > int(md.AntennaIndex) {
```
Comment (johanstokking) [inline pkg/gatewayserver/io/io.go]: Where does this come from? 

Should we be getting `antennnas` here?

https://github.com/TheThingsNetwork/lorawan-stack/blob/96ebc355c10a456224f1957bf494c10f1fef38f3/pkg/gatewayserver/gatewayserver.go#L361-L371

Also, make sure that `antennas` is returned when GS uses cluster authentication (i.e. in an unauthenticated gateway context, when the gateway uses UDP for example).

File: pkg/gatewayserver/io/io.go (hunk 1 above)
Comment (neoaggelos) [reply pkg/gatewayserver/io/io.go]: Purpose is to pass a copy of the antenna location value. Is it safe to just pass a reference?

File: pkg/gatewayserver/io/io.go (hunk 2 above)
Comment (neoaggelos) [reply pkg/gatewayserver/io/io.go]: Should be OK now

**Comment by johanstokking** [comment]
The open issue is that when gateways link with an API key that has `INFO` rights, GS gets the location even when `location_public` is false, and GS will make it part of the `RxMetadata`. This is an edge case; gateways should not be linking with rights other than `LINK`, but we should account for this. Because users.

So GS also has to get the `location_public` setting, and if that is false, strip the location.

**Comment by neoaggelos** [comment]
GS now honors the `location_public` flag. We still need to add a few tests for this before it can be merged.

File: pkg/gatewayserver/gatewayserver.go [hunk 3]
```diff
@@ -362,8 +362,10 @@ func (gs *GatewayServer) Connect(ctx context.Context, frontend io.Frontend, ids
 		GatewayIdentifiers: ids,
 		FieldMask: pbtypes.FieldMask{
 			Paths: []string{
+				"location_public",
 				"downlink_path_constraint",
 				"enforce_duty_cycle",
+				"antennas",
```
Comment (johanstokking) [inline pkg/gatewayserver/gatewayserver.go]: Keep this list sorted ascending

**Pull request merged**